packages/code-editor/src/testing/getTestUtils.spec.tsx

```diff
@@ -0,0 +1,479 @@
+/**
+ * CodeEditor Test Utils Tests & Consumer Examples
+ *
+ * This file serves dual purposes:
+ * 1. Tests the getTestUtils implementation to ensure it works correctly
+ * 2. Provides examples for consumers on how to use getTestUtils in their own tests
+ *
+ * KEY PATTERNS FOR CONSUMERS:
+ *
+ * 1. Always wait for initialization:
+ *    const utils = getTestUtils('your-lgid');
+ *    await utils.waitForInitialization();
+ *
+ * 2. Wait for loading to complete when testing loading states:
+ *    await utils.waitForLoadingToComplete();
+ *
+ * 3. Use async/await for getContent():
+ *    const content = await utils.getContent();
+ *
+ * 4. Test actual values, not just types:
+ *    expect(utils.getIsLoading()).toBe(false); // Not just typeof
+ *
+ * See the "Integration Tests" section below for complete consumer examples.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import '@testing-library/jest-dom';
+
+import { CodeEditor } from '../CodeEditor';
+import { LanguageName } from '../CodeEditor/hooks/extensions/useLanguageExtension';
+import { Panel } from '../Panel';
+
+import { getTestUtils } from './getTestUtils';
+
+// For testing purposes, create a simplified test environment that focuses on
```

Comment:
Similar to what Adam mentioned in dev sync yesterday, should we be testing the actual CodeEditor component and not a mock?